lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -37,6 +37,19 @@ defmodule Kernel.ErrorsTest do
       'Foo.:"\#{:bar}"'
   end
 
+  test :invalid_identifier do
+    assert_compile_fail SyntaxError, "nofile:1: unexpected token: foo@", 'foo@'
```

Comment:
Oops, I was not clear, I think the unexpected token should only be `@`. But now that I think about it, it is better to show the whole thing. Maybe we should go with "syntax error before: foo@`?